src/client.ts

```diff
@@ -1,124 +1,154 @@
+      retryDelay: (retryCount, error) => {
+        if (error.response?.status === 429) {
+          return retryCount * 1000;
+        }
+
+        return 0;
+      },
+    });
+  }
 
   public async verifyAuthentication(): Promise<void> {
-    // TODO make the most light-weight request possible to validate
-    // authentication works with the provided credentials, throw an err if
-    // authentication fails
-    const request = new Promise<void>((resolve, reject) => {
-      http.get(
-        {
-          hostname: 'localhost',
-          port: 443,
-          path: '/api/v1/some/endpoint?limit=1',
-          agent: false,
-          timeout: 10,
-        },
-        (res) => {
-          if (res.statusCode !== 200) {
-            reject(new Error('Provider authentication failed'));
-          } else {
-            resolve();
-          }
-        },
-      );
+    const endpoint = `${BASE_URI}/corps`;
+
+    await this.fetch(endpoint, {
+      headers: this.generateHeaders(),
     });
+  }
 
-    try {
-      await request;
-    } catch (err) {
-      throw new IntegrationProviderAuthenticationError({
-        cause: err,
-        endpoint: 'https://localhost/api/v1/some/endpoint?limit=1',
-        status: err.status,
-        statusText: err.statusText,
-      });
+  /**
+   * Iterates each corp resource in the provider.
+   * Note: Pagination is not currently supported on this endpoint by the provider.
+   *
+   * @param iteratee receives each resource to produce entities/relationship
+   */
+  public async iterateCorps(
+    iteratee: ResourceIteratee<SignalSciencesCorp>,
+  ): Promise<void> {
+    const endpoint = `${BASE_URI}/corps`;
+
+    const { data } = await this.fetch(endpoint, {
+      headers: this.generateHeaders(),
+    });
+
+    for (const corp of data) {
+      await iteratee(corp);
     }
   }
 
   /**
-   * Iterates each user resource in the provider.
+   * Iterates each user resource in the provider based on the provided corp.
+   * Note: Pagination is not currently supported on this endpoint by the provider.
    *
    * @param iteratee receives each resource to produce entities/relationships
    */
   public async iterateUsers(
-    iteratee: ResourceIteratee<AcmeUser>,
+    corpName: string,
+    iteratee: ResourceIteratee<SignalSciencesUser>,
   ): Promise<void> {
-    // TODO paginate an endpoint, invoke the iteratee with each record in the
-    // page
-    //
-    // The provider API will hopefully support pagination. Functions like this
-    // should maintain pagination state, and for each page, for each record in
-    // the page, invoke the `ResourceIteratee`. This will encourage a pattern
-    // where each resource is processed and dropped from memory.
-
-    const users: AcmeUser[] = [
-      {
-        id: 'acme-user-1',
-        name: 'User One',
-      },
-      {
-        id: 'acme-user-2',
-        name: 'User Two',
-      },
-    ];
+    const endpoint = this.buildEndpoint(corpName, '/users');
```

Comment:
Is this endpoint paginated? If so, can you make sure that `iterateUsers` checks for a `nextPage`?

Reply:
A number of SigSci's endpoints are paginated. Following the pattern there, I tried to paginated `/users` and `/corps` endpoints with no behavioral changes. If SigSci adds pagination in the future, we will want to implement it. There is a note in client.ts about this.

Reply:
> * Note: Pagination is not currently supported on this endpoint by the provider.

Yes, I should've read the docstrings! 🙇